Source/Core/DolphinQt/DiscordHandler.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#ifdef USE_DISCORD_PRESENCE
+
+#include "DolphinQt/DiscordHandler.h"
+
+#include <iterator>
+
+#include <QApplication>
+
+#include "Common/Thread.h"
+
+#include "UICommon/DiscordPresence.h"
+
+#include "DolphinQt/DiscordJoinRequestDialog.h"
+
+DiscordHandler::DiscordHandler(QWidget* parent) : QObject{parent}, m_parent{parent}
+{
+  connect(this, &DiscordHandler::JoinRequest, this, &DiscordHandler::ShowNewJoinRequest);
+}
+
+DiscordHandler::~DiscordHandler()
+{
+  Stop();
+}
+
+void DiscordHandler::Start()
+{
+  m_stop_requested.Set(false);
+  m_thread = std::thread(&DiscordHandler::Run, this);
+}
+
+void DiscordHandler::Stop()
+{
+  m_stop_requested.Set(true);
+
+  if (m_thread.joinable())
+    m_thread.join();
+}
+
+void DiscordHandler::DiscordJoinRequest(const char* id, const std::string& discord_tag,
+                                        const char* avatar)
+{
+  m_request_dialogs.emplace_front(m_parent, id, discord_tag, avatar);
```

Comment:
This is not thread safe, as it creates a new QDialog. It should be moved inside the signal handler.

Reply:
Sorry but I didn't see this message before pushing the change, I'll change the ``discord_tag`` to be a std::string isn't a std::string&